src/Pact/Analyze/Translate.hs

```diff
@@ -1573,6 +1573,10 @@ runTranslation modName funName info caps pactArgs body checkType = do
               CheckDefpact ->
                 withNewScope (PactScope modName funName) bindingTs $
                   Some SStr . Pact <$> translatePact body
+              CheckPactStep ->
+                -- TODO: use withNewStep?
+                withNewScope StepScope bindingTs $
```

Comment:
I don't remember the motivation for `withNewStep` well enough to remember if we should use it here.

Reply:
It looks like `withNewStep` does not do `fmap (mapExistential (Return tid))` over `act`. The `Return` is necessary for returning values from scopes. Also `withNewStep` doesn't take bindings (at least currently).